### PR TITLE
chore: centralised cdk scripts

### PR DESCRIPTION
**Pull request by joshbalfour**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #5656

### CODE REVIEW

File: packages/deployment-service/cdk/cdk.ts [hunk 1]
```diff
@@ -12,12 +11,15 @@ const bootstrap = async () => {
     cwd: path.resolve(__dirname, '..'),
   })
 
+  const zipPath = path.resolve(__dirname, '..', 'dist', 'main.zip')
+
+  await exec(`rm -rf ${zipPath}`)
```
Comment (bashleigh) [inline packages/deployment-service/cdk/cdk.ts:16]: <3

File: packages/ts-scripts/src/cdk/components/api-gateway-api.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const authorizer = getAuthorizer(scope, cognitoUserPoolId)
+    defaultMethodOptions = {
+      authorizationType: apigateway.AuthorizationType.COGNITO,
+      authorizer,
+    }
+  }
+
+  const defaultCorsPreflightOptions = allowCors ? {
+    allowOrigins: ['*'],
+    allowHeaders: ['Content-Type', 'Authorization', 'X-Api-Key', 'api-version'],
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/api-gateway-api.ts:26]: should we make these options?

File: packages/ts-scripts/src/cdk/components/codebuild-project.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }),
+  })
+  
+  return project
+}
+
+export const getCodebuildSnsTopic = (stack: cdk.Stack): Topic => {
+  const topic = createSnsTopic(stack, 'codebuild-sns-topic')
+
+  const rule = new Rule(stack, 'ccodebuild-sns-topic-rule', {
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/codebuild-project.ts:26]: Made a suggestion but might need to be an option? 
```suggestion
  const rule = new Rule(stack, 'codebuild-sns-topic-rule', {
```

File: packages/ts-scripts/src/cdk/components/cognito-authorizer.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    authorizers.set(userpoolId, new CognitoUserPoolsAuthorizer(scope, 'authorizer', {
+      cognitoUserPools: [UserPool.fromUserPoolId(scope, 'user-pool-authorizer', userpoolId)],
+    }))
+  }
+  const authorizer = authorizers.get(userpoolId)
+  if (!authorizer) {
+    throw new Error('Authorizer not found')
+  }
+  return authorizer
+}
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/cognito-authorizer.ts:18]: EOF
```suggestion
}

```

File: packages/ts-scripts/src/cdk/components/iam-policy.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export {
+  PolicyStatement,
+  Effect,
+} from '@aws-cdk/aws-iam'
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/iam-policy.ts:4]: EOF
```suggestion
} from '@aws-cdk/aws-iam'

```

File: packages/ts-scripts/src/cdk/components/rds-database.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    cloudwatchLogsRetention: RetentionDays.ONE_MONTH,
+  })
+
+  db.connections.allowFromAnyIpv4(ec2.Port.allTcp())
+
+  return db
+}
+
+export type DatabaseCluster = rds.DatabaseCluster
+export type ISecret = secretsmanager.ISecret
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/rds-database.ts:23]:
```suggestion
export type ISecret = secretsmanager.ISecret

```

File: packages/ts-scripts/src/cdk/components/sns-topic.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import * as sns from '@aws-cdk/aws-sns'
+import * as cdk from '@aws-cdk/core'
+
+export const createSnsTopic = (stack: cdk.Stack, name: string): Topic => {
+  return new sns.Topic(stack, name)
+}
+
+export type Topic = sns.Topic
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/sns-topic.ts:8]:
```suggestion
export type Topic = sns.Topic

```

File: packages/ts-scripts/src/cdk/components/sqs-queue.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+export const createSqsQueue = (stack: cdk.Stack, name: string, visibilityTimeout?: number): sqs.Queue => {
+  return new sqs.Queue(stack, name, {
+    visibilityTimeout: cdk.Duration.seconds(visibilityTimeout || 30),
+  })
+}
+
+export {
+  Queue
+} from '@aws-cdk/aws-sqs'
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/sqs-queue.ts:12]:
```suggestion
} from '@aws-cdk/aws-sqs'

```

File: packages/ts-scripts/src/cdk/components/stack.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  component: string
+}): cdk.Stack => {
+  const scope = new cdk.App()
+  const name = `${namespace}-${appName}-${component}`
+  const stack = new cdk.Stack(scope, name)
+
+  return stack
+}
+
+export type Stack = cdk.Stack
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/stack.ts:19]:
```suggestion
export type Stack = cdk.Stack

```

File: packages/ts-scripts/src/cdk/components/vpc-network.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import * as ec2 from '@aws-cdk/aws-ec2'
+import * as cdk from '@aws-cdk/core'
+
+export const createVpc = (scope: cdk.Stack, name: string): ec2.Vpc => {
+  const vpc = new ec2.Vpc(scope, name)
+
+  return vpc
+}
+
+export type Vpc = ec2.Vpc
```
Comment (bashleigh) [inline packages/ts-scripts/src/cdk/components/vpc-network.ts:10]:
```suggestion
export type Vpc = ec2.Vpc

```

**Review by bashleigh — CHANGES_REQUESTED** [review]
Just a few questions and suggestions

File: packages/ts-scripts/src/cdk/components/codebuild-project.ts (hunk 3 above)
Comment (joshbalfour) [reply packages/ts-scripts/src/cdk/components/codebuild-project.ts]: there can only be one per stack so it's fine to be hard coded

File: packages/deployment-service/cdk/lib/create-policies.ts [hunk 11]
```diff
@@ -1,9 +1,9 @@
-import { Project } from '@aws-cdk/aws-codebuild'
-import { Effect, PolicyStatement } from '@aws-cdk/aws-iam'
-import { DatabaseCluster } from '@aws-cdk/aws-rds'
-import { Bucket } from '@aws-cdk/aws-s3'
-import { ISecret } from '@aws-cdk/aws-secretsmanager'
-import { Queue } from '@aws-cdk/aws-sqs'
+import { Project } from '@reapit/ts-scripts/src/cdk/components/codebuild-project'
```
Comment (willmcvay) [inline packages/deployment-service/cdk/lib/create-policies.ts:1]: Can you not add the CDK stuff to the ts-scripts manifests so you don't have to provide the full path? No worries if not, just a thought

**Review by willmcvay — APPROVED** [review]
Just a small thing but LGTM

**Pull request merged**